### PR TITLE
Draft spec for theming, Mica

### PR DESCRIPTION
**Pull request by zadjii-msft**
⚠️ This spec is going into the `specs/drafts/` folder, because it's CLEARLY not done yet.

I discussed this a bit with Dustin. We felt it would be valuable to have these thoughts committed as a durable artifact. Better to have our Mica thoughts written down somewhere, with the context they belong in. That of course includes the bigger Theming spec, which never got finished. 

I don't think we need to go through the fill spec review for this. Theming is clearly still a WIP. But committing this draft should give a better picture of what the vision is. 

See also: 
* #3327
* #10509

### TODOs
* [ ] The many that are straight up in the doc
* [ ] The fact that there's multiple Mica's now
* [ ] GO look at MSFT:39027976

### CODE REVIEW

**Comment by miniksa** [comment]
Is there any chance of getting improved platform support for some of these limitations of Mica? I didn't see that called out anywhere. You say that we should make it ergonomic for our users to choose things here, but the platform is definitely making this hard.

**Comment by zadjii-msft** [comment]
> the platform is definitely making this hard

I'm under the impression that's _by design_. The way the platform seems to be designed prevents Mica from being used in other places where it's not really part of the design language. Like, perhaps adressing the fact that apps would just use acrylic wherever they wanted, multiple times in the same app, against the official guidance, which played into the performance concerns with acrylic. \</baseless conjecture>

Part of marking this as a draft is to get this out into the world as our baseline expectations of Mica in the Terminal, to help communicate our goals with the rest of the platform team 😉

File: doc/specs/drafts/#3327 - Application Theming/#10509 - Mica.md [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+  It however, did not seem to work for the "drag window", the child HWND which
+  we use to intercept nonclient messages in our titlebar area. Apparently, that
+  API does not work at all for `WS_CHILD` windows, by design. This unfortunately
+  prevents us from allowing Mica only in the titlebar area, without also
+  applying it to the rest of the main window.
+* We considered rolling our own Mica brush to allow us to style individual
+  elements. This didn't seem terribly hard at face value, considering [the Mica
+  recipe] is basically open-source. However, it seems to rely on some
+  closed-source `IGraphicsEffect`s from the `Composition.Effects` namespace.
+  Apparently, those effects are only [implemented in Win2D]. It's unclear at
```
Comment (sylveon) [inline doc/specs/drafts/#3327 - Application Theming/#10509 - Mica.md:144]: These effects are implemented in Direct2D, and Win2D is just a thin wrapper over Direct2D. It's possible to write your own classes for these Direct2D effects to be able to use them without needing to import the whole of Win2D. This is actually what WinUI does to use these effects without making it depend on Win2D.

Comment (zadjii-msft) [reply doc/specs/drafts/#3327 - Application Theming/#10509 - Mica.md:148]: I mean yea, we probably could do that. For ex:
https://github.com/microsoft/Win2D/blob/da3fa1e6caab9df443c43d0f7a211d9d159539f8/winrt/lib/effects/generated/ColorSourceEffect.cpp

**Comment by zadjii-msft** [comment]
Note to reviewers: Probably read through at least "**Highest priority theming properties**", and then the sections titled "**Theme Colors**", "**Titlebar complications**" and "**Light & dark mode theming**". That's probably enough to get the gist here.

**Review by lhecker — COMMENTED** [review]
The only critique I have is:
I would strongly prefer `#rrggbbaa` (and `rgba(r, g, b, a)`) over `#aarrggbb`. The first two are in line with the [CSS Color Module Level 4 draft](https://www.w3.org/TR/css-color-4/#hex-notation) and supported by all major browsers. I have a parser for `rgba(r, g, b, a)` at hand as well which I can share with you. 🙂

Some unimportant thoughts:
* Should themes get UUIDs?
* When the spec says `window.background.useMica`, the first dot denotes a sub-object while the second dot is meant literally - I think this is a bit confusing at times and it makes me wonder whether we should make something like `window.background` a sub-object as well.
* Extendability always makes me a bit afraid that we're biting off more than we can chew / that we'll have a hard time keeping all features not just there but also stable and bug free with the current team size we have. This of course doesn't apply much to theming properties which can be 1:1 propagated to WinUI properties, but... Well you know what I mean...

**Review by lhecker — APPROVED** [review]

**Comment by zadjii-msft** [comment]
Hey this is a DRAFT spec, not even a final version, so I've got a mind to merge this with 2 instead of 3. When we undraft the spec, we can go for 3 again.

I'll give the bot a few hours for any NAKs.

@msftbot merge this in 8 hours

**Pull request merged**